src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -100,29 +100,106 @@ export class DeploymentHelper {
+        const build: asa.Build = {
+            properties: {
+                relativePath: uploadResponse.relativePath,
+                builder: params.builder ? `${resourceId}/buildServices/${buildServiceName}/builders/${params.builder}` : `${resourceId}/buildServices/${buildServiceName}/builders/default`,
+                agentPool: `${resourceId}/buildServices/${buildServiceName}/agentPools/default`,
+            }
+        };
+        let transformedBuildEnvironmentVariables = {};
+        if (params.buildEnv) {
+            core.debug("Build environment variables modified.");
+            const parsedBuildEnvVariables = parse(params.buildEnv);
+            //Parsed pairs come back as  {"key1":{"value":"val1"},"key2":{"value":"val2"}}
+            Object.keys(parsedBuildEnvVariables).forEach(key => {
+                transformedBuildEnvironmentVariables[key] = parsedBuildEnvVariables[key]['value'];
+            });
+            build.properties.env = transformedBuildEnvironmentVariables;
+        }
+        core.debug('build: ' +  JSON.stringify(build));
+        const buildResponse = await client.buildServiceOperations.createOrUpdateBuild(params.resourceGroupName, params.serviceName, buildServiceName, buildName, build);
+        core.debug('build response: ' +  JSON.stringify(buildResponse));
+        const regex = RegExp("[^/]+$");
+        const buildResultName = regex.exec(buildResponse.properties.triggeredBuildResult.id)[0];
+        let buildProvisioningState = 'Queuing';
+        let cnt = 0;
+        core.debug("wait for build result......");
+        //Waiting for build result. Timeout 30 minutes.
+        while (buildProvisioningState != 'Succeeded' && buildProvisioningState != 'Failed' && cnt++ < 180) {
+            await new Promise(resolve => setTimeout(resolve, 10000));
+            console.log("wait for 10 seconds....");
+            const waitResponse = await client.buildServiceOperations.getBuildResult(params.resourceGroupName, params.serviceName, buildServiceName, buildName, buildResultName);
+            core.debug('build result response: ' + JSON.stringify(waitResponse));
+            buildProvisioningState = waitResponse.properties.provisioningState;
+        }
+        if (cnt == 180) {
+            throw Error("Build result timeout.");
+        }
+        if (buildProvisioningState == 'Failed') {
+            throw Error("Build result failed.");
+        }
+        let deploymentResource: asa.DeploymentResource = await this.buildDeploymentResource(client, params, sourceType, buildResponse.properties.triggeredBuildResult.id);
+        core.debug("deploymentResource: " + JSON.stringify(deploymentResource));
+        const deployResponse = await client.deployments.beginCreateOrUpdateAndWait(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName, deploymentResource);
+        core.debug('deploy response: ' + JSON.stringify(deployResponse));
```

Comment:
Will we expose the environment variables to the logs here? If yes, we need to remove them, or we should avoid dumping the whole payload to the console.

Reply:
It will only dump the deployment resource payload to the console when debug mode on and will hide the secret info. I think it may help if some error happens. 
![image](https://user-images.githubusercontent.com/86712051/190543545-94a54aa8-bc7d-4f93-ac18-8f4b6a04e7a6.png)